src/modules/previewpane/SvgThumbnailProvider/SvgThumbnailProvider.cs

```diff
@@ -118,92 +117,155 @@ public Bitmap GetThumbnailImpl(uint cx)
             _browser.Height = (int)cx;
             _browser.NavigationCompleted += async (object sender, CoreWebView2NavigationCompletedEventArgs args) =>
             {
-                var a = await _browser.ExecuteScriptAsync($"document.getElementsByTagName('svg')[0].viewBox;");
-                if (a != null)
+                try
                 {
-                    await _browser.ExecuteScriptAsync($"document.getElementsByTagName('svg')[0].style = 'width:100%;height:100%';");
-                }
+                    // Check if the SVG element is present
+                    var a = await _browser.ExecuteScriptAsync($"document.getElementsByTagName('svg')[0].viewBox;");
+                    if (a != null)
+                    {
+                        var svgContent = SvgContents.Substring(SvgContents.IndexOf("<svg", StringComparison.OrdinalIgnoreCase), SvgContents.IndexOf("</svg>", StringComparison.OrdinalIgnoreCase) - SvgContents.IndexOf("<svg", StringComparison.OrdinalIgnoreCase) + "</svg>".Length);
+
+                        Dictionary<string, string> styleDict = new Dictionary<string, string>();
+
+                        // Try to parse the SVG content
+                        try
+                        {
+                            // Attempt to parse the svgContent
+                            var svgDocument = XDocument.Parse(svgContent);
+                            var svgElement = svgDocument.Root;
+                            var currentStyle = svgElement?.Attribute("style")?.Value;
+
+                            // If style attribute exists, preserve existing styles
+                            if (!string.IsNullOrEmpty(currentStyle) && currentStyle != "null")
```

Comment:
The loop that constructs styleDict could be refactored for readability. Using LINQ to filter and map values directly could make the code shorter and clearer.
 Consider using a LINQ query to process currentStyle.Split(';'), which could simplify the code and make it easier to maintain.